layers/README.md

```diff
@@ -0,0 +1,85 @@
+| Layer().export_file(_input_) | Saves the current state of the loaded ATT&CK layer to a json file denoted by the _input_. |
+| Layer().get_dict() | Returns a representation of the current ATT&CK layer object as a dictionary. | 
+
+#### Example Usage
+
+```python
+example_layer_dict = {
+    "name": "example layer",
+    "version": "3.0",
+    "domain": "mitre-enterprise",
+    "description": "hello, world",
+    ...
+    }
+example_layer_location = "/path/to/layer/file.json"
+example_layer_out_location = "/path/to/new/layer/file.json"
+
+from layers.core.layer import Layer
+
+layer1 = Layer(example_layer_dict)
+layer1.export_file(example_layer_out_location)
+
+layer2 = Layer()
+layer2.load_input(example_layer_dict)
+layer2.get_dict()
+
+layer3 = Layer()
+layer3.load_file(example_layer_location)
+```
+
+## layerops.py
+Layerops.py provides the LayerOps class, which is a way to combine layer files in an automated way, using user defined lambda functions. Each LayerOps instance, when created, ingests the provided lambda functions, and stores them for use. An existing LayerOps class can be used to combine layer files according to the initialized lambda using the process method. The breakdown of this two step process is documented in the table below, while examples of both the list and dictionary modes of operation can be found below.
+
+| method | description |
+|:-------|:------------|
+| LayerOps(score=_score_, comment=_comment_, enabled=_enabled_, colors=_colors_, metadata=_metadata_, name=_name_, desc=_desc_, default_values=_default_values_) | Each of the _inputs_ takes a lambda function that will be used to combine technique object fields matching the parameter. The one exception to this is _default_values_, which is an optional dictionary argument containing default values to provide the lambda functions if elements of the combined layers are missing them. |
+| LayerOps.process(_data_, defaults=_defaults_) | Applies the lambda functions stored during initialization to the layer objects in _data_. _data_ must be either a list or a dictionary of Layer objects, and is expected to match the format of the lambda equations provided during initialization. |
+
+#### Example Usage
+```python
+from layers.manipulators.layerops import LayerOps
+from layers.core.layer import Layer
+
+demo = Layer()
+demo.load_file("C:\Users\attack\Downloads\layer.json")
+demo2 = Layer()
+demo2.load_input({"name": "example layer", ... })
+
+lo = LayerOps(score=lambda x: x[0] * x[1], 
+              name=lambda x: x[1], 
+              desc=lambda x: "This is an list example")
+out_layer = lo.process([demo, demo2])
+out_layer.export_file("C:\demo_layer1.json")
+
+lo2 = LayerOps(score=lambda x: x['a'], 
+               color=lambda x: x['b'], 
+               desc=lambda x: "This is a dict example")
+out_layer2 = lo2.process({'a': demo, 'b': demo2})
+dict_layer = out_layer2.get_dict()
+print(dict_layer)
+out_layer2.export_file("C:\demo_layer2.json")
+```
```

Comment:
We need additional examples here and better explanations about what the existing examples do.

For the additional examples, we should have actual applications of how layerops might be used. Some possibilities:
- average scores of any number of input layers
- reverse the score of the input layer (`100-x[0]` assuming the score is on a 0-100 scale)
- Merge comments (`"; ".join(x)` I think?)

Each example should have a comment explaining what that layerops instance's purpose is.

We should also demonstrate the functionality of a single layerops instance being reused. For example, show the "average layers" being used on layers A,B, and then C,D,E in two separate executions.

Reply:
I've got examples for each of these. Let me know what you think.